src/test/shell/bazel/starlark_repository_test.sh

```diff
@@ -2704,4 +2704,108 @@ EOF
   expect_log "I see: @@bar~override//:data"
 }
 
+function test_file_watching_inside_working_dir() {
```

Comment:
nit: maybe add a test that watches the same file twice? For even more creativity: once through a label, once through an absolute path?

Reply:
> maybe add a test that watches the same file twice?

funnily enough, all existing tests (as in, not the ones I added) already do that because they read a file pointed to by a label (`rctx.read(Label("//:something"))`). This watches the label once when it's converted to a path, and another time when it's read by `rctx.read`.

> For even more creativity: once through a label, once through an absolute path?

Anything addressable by a label can only be watched as a label, so I don't know if that makes sense.

Reply:
Where is it enforced that anything addressable by a label can only be watched as a label? AFAICT the new code in `toRootedPath()` would happily do its thing if one called `rctx.watch($OUTPUT_BASE + "/external/reponame/repopath")`? Maybe this is where our misunderstanding on the thread in that method comes from?

Reply:
yes, it "happily does its thing" if you called `rctx.watch($OUTPUT_BASE + "/external/reponame/repopath")`, and per the logic at https://github.com/bazelbuild/bazel/pull/21230/files#diff-3d5c46900301b3c4ed255b0011208cd31510a4ff2065ad15f26398dd8761d4c7R1206, it would be the same as if you called `rctx.watch(Label("@@reponame//:repopath"))`. Note how anything inside the workspace (`FileInsideWorkspace`) is "watched" using the "repo+path" mechanism, instead of the absolute path mechanism (which is only for `FileOutsideWorkspace`).

Reply:
Oh, I see. It would be better if we could make the rule that files under the control of Bazel can only be referenced by the Label syntax, but I guess that ship has already sailed...